XenAdmin/Wizards/PatchingWizard/PatchingWizard_SelectServers.cs

```diff
@@ -148,6 +146,26 @@ protected override void PageLoadedCore(PageLoadedDirection direction)
                         int index = dataGridViewHosts.Rows.Add(new PatchingHostsDataGridViewRow(host, hasPool, !poolSelectionOnly));
                         EnabledRow(host, SelectedUpdateType, index);
                     }
+
+                    if (hasPool)
+                    {
+                        if (poolIndex == -1)
+                            continue;
+                        var poolRow = (PatchingHostsDataGridViewRow) dataGridViewHosts.Rows[poolIndex];
+                        var masterRow = (PatchingHostsDataGridViewRow) dataGridViewHosts.Rows[poolIndex + 1];
+                        poolRow.Enabled = false;
+                        for (int index = poolIndex + 1; index < dataGridViewHosts.RowCount; index ++)

```

Comment:
We should avoid iterating through the host rows again and instead enable/disable the pool row in the previous loop (when we create the host rows):
- when we add the pool row, disable it, `poolRow.Enabled = false`;
- when we add a host row that is enabled, we should also enable the pool row; 
- after we added all the host rows and the pool row is still disabled, then set its ToolTipText to the master's tooltip text (master row is `dataGridViewHosts.Rows[dataGridViewHosts.Rows.IndexOf(poolRow) + 1]`)